aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java

```diff
@@ -177,6 +181,22 @@ public void validatePutObject() {
         PutObjectRequest.builder().bucket(BUCKET).key("some/key").build(), Paths.get("/etc/hosts"));
   }
 
+  @Test
+  public void validateDeleteObjects() {
+    Path sourcePath = Paths.get("/etc/hosts");
+    s3.putObject(PutObjectRequest.builder().bucket(BUCKET).key("some/key1").build(), sourcePath);
+    s3.putObject(PutObjectRequest.builder().bucket(BUCKET).key("some/key2").build(), sourcePath);
+
+    Delete objectsToDelete =
+        Delete.builder()
+            .objects(
+                ObjectIdentifier.builder().key("some/key1").build(),
+                ObjectIdentifier.builder().key("some/key2").build())
+            .build();
+
+    s3.deleteObjects(DeleteObjectsRequest.builder().bucket(BUCKET).delete(objectsToDelete).build());
```

Comment:
I've verified this integration test exercises this new code path via the debugger, but maybe we can do better in terms of test hardness (i.e. validate on the server side that for such requests we actually get the body). Lmk your thoughts @nastra @danielcweeks

Reply:
Good idea, I think for purpose of testing we can have an internal class `TestS3SignerServlet` which extends the `S3SignerServlet`. This servlet implementation can accept a set of precondition predicates on the expected request state (in our case, for POST ?delete do we actually have a body).

This implementation can override the normal post/get/put/head implementations in the following manner:

1.) Verify the preconditions
2.) Delegate to parent super.post()/get/etc

Reply:
I just opted for adding this validation logic in the S3Servlet itself since that's just part of tests. Extending the servlet proves to be challenging because we want to inspect the body but the body input stream would need to be read twice, which requires a lot of complication for what we're trying to achieve. Lmk what you think @nastra